src/librustc/hir/lowering.rs

```diff
@@ -5187,6 +5217,10 @@ impl<'a> LoweringContext<'a> {
         }
     }
 
+    fn arg(&mut self, hir_id: hir::HirId, pat: P<hir::Pat>, source: hir::ArgSource) -> hir::Arg {
+        hir::Arg { hir_id, pat, source }
```

Comment:
`hir_id: self.next_id()` would probably be more ergonomic.

Reply:
This function is given the `HirId` of the original argument rather than a newly generated one.

Reply:
Ah, then maybe don't add this method?

Reply:
I've removed it.